specs/donut_and_sparkline_chart_component.adoc

```diff
@@ -0,0 +1,197 @@
+                <div class="list-view-pf-additional-info-item donut-chart" chart-title="Storage" chart-data="node.chartData">
+                </div>
+            </div>
+        </div>
+
+        NOTE: if "vm.chartData.percent-used" is not defined . than automatically "donut chart" will show "no data available".
+
+	*** chartThresholds (optional): If user will not pass thresholds value as parameter 	than it will have default value: {'warning':'60','error':'90'} which should be 
+		defined in "config.json". because it should be configurable.
+		Example: -
+		
+		(In controller):
+
+		vm.chartThresholds = {'warning':'40','error':'90'}; (defined chartThresholds object)
+
+		(In your template):
+
+		<div ng-repeat="host in hostList"
+            <div  class="col-md-2">
+                <div class="list-view-pf-additional-info-item donut-chart" chart-title="Storage" chart-data="node.chartData" chart-thresholds="chartThresholds">
+                </div>
+            </div>
+        </div>
+		
+
+	* Sparkline chart related changes:
+
+	** We are going to create "angular directive" for Sparkline chart component in order to draw "Sparkline chart". and directive name will be "sparklineChart" and it will support the following parameters.
+
+	*** chartId: Expected "unique number" for each chart in one template.
+				For example: if you are using ng-repeat .
+
+		 
+		<div ng-repeat="host in hostList"> <sparkline-chart chart-id="{{$index}}"> </sparkline-chart> </div>
+		
+
+		NOTE: Internally we are using "c3" Chart (plugin used for draw chart) so it will expect unique id for each chart.
+
+	*** chartData: Should be one array with used value with their corresponding timestamp.
+		Example: -
+		
+        [["usedValue1","timestamp1"],["usedValue2","timestamp2"],["usedValue3","timestamp3"]]
+        
+        (In your controller):
+
+        vm.chartData = [
+                [10, 1311836001],
+                [20, 1311836002],
+                [30, 1311836003],
+                [20, 1311836004],
+                [30, 1311836005],
+                [10, 1311836006]
+        ]
+
+        (In your template):
+
+        <div ng-repeat="host in hostList"
+            <sparkline-chart chart-id="{{$index}}" chart-data="chartData">
+            </sparkline-chart>
+        </div>
+
+        
+
+        NOTE: if your "vm.chartData" is empty array. than automatically "sparkline chart" will show "no data available".
+
+
+
+=== Alternatives
+
+We can use static c3 chart plugin without any angular wrapper(angular patternfly). but it will be very difficult to handle in angular application.
+
+=== Data model impact:
+
+See 'Impacted Modules' section where all the details are populated.
+
+=== Impacted Modules:
+
+==== Tendrl frontend impact:
+
+   * We need to add c3js chart (http://c3js.org/) libaray . and also one wrapper "angular patternfly" (http://angular-patternfly.rhcloud.com/) in order to access c3 chart features in angularjs application.
```

Comment:
No . not required .. automatically importing node_modules will take care for build/spec as well.